internal/schema/types.go

```diff
@@ -1,7 +1,12 @@
 package schema
 
+import "regexp"
+
 // AssetType define an asset type
-type AssetType string
+type AssetType struct {
+	Name             string         `json:"name"`
+	ValidationRegexp *regexp.Regexp `json:"-"`
```

Comment:
Personally, I would have used a Validator interface or a functor or a list of them because validation could be something else than a regexp (for instance a date should be valid in the calendar or an int could be in a range, this is harder to implement with regexps). However, you can still provide the regexp validator as a utility in the package for our specific current need.

Reply:
Agreed, done